core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -61,8 +66,7 @@ void setSurfaceHolderRequester(SurfaceHolderRequester surfaceHolderRequester) {
 
     void prepareVideo(final Uri videoUri) {
         if (surfaceHolderRequester == null) {
-            logPlayerNotAttachedWarning("prepareVideo()");
-            return;
+            throw new IllegalStateException("Must set a SurfaceHolderRequester before preparing video");
```

Comment:
the surface is attached by a view, it's not guaranteed. Eventually when we support audio only we'll have a separate flow